### DISCUSSION TITLE
GITHUB_TOKEN replacement

### DISCUSSION THREAD

**Discussion by aleks-ivanov** [community/community#27067] [Actions]
<p>Hi!</p>
<p>I want to replace the GITHUB_TOKEN for a more specific one and there is one thing I can’t understand.</p>
<p>For the three actions that I am using, that require GITHUB_TOKEN, a token with only “public_repo” permission is enough, but “public_repo” does include main permissions in itself.</p>
<p>Is there a way to specify even more or that’s the lowest level of specificity currently ?</p>

**Comment by airtower-luna** [comment, marked as answer]
<p>Options to restrict the <code>GITHUB_TOKEN</code> have been added recently:</p>
<aside class="onebox allowlistedgeneric" data-onebox-src="https://github.blog/changelog/2021-04-20-github-actions-control-permissions-for-github_token/">
  <header class="source">
      <img src="https://github.blog/wp-content/uploads/2019/01/cropped-github-favicon-512.png?fit=32%2C32" class="site-icon" width="32" height="32">

      <a href="https://github.blog/changelog/2021-04-20-github-actions-control-permissions-for-github_token/" target="_blank" rel="noopener">The GitHub Blog</a>
  </header>

  <article class="onebox-body">
    <div class="aspect-image" style="--aspect-ratio:690/195;"><img src="https://user-images.githubusercontent.com/5814919/181101561-2e66645c-2de2-4cf0-8d5a-919172cc7391.png" class="thumbnail" width="690" height="195" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/e/5/e5c1b9498b0d4932c74ba3d8f1e8092358b5b92e_2_690x195.png, https://aws1.discourse-cdn.com/github/optimized/3X/e/5/e5c1b9498b0d4932c74ba3d8f1e8092358b5b92e_2_1035x292.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/3X/e/5/e5c1b9498b0d4932c74ba3d8f1e8092358b5b92e_2_1380x390.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/e/5/e5c1b9498b0d4932c74ba3d8f1e8092358b5b92e_2_10x10.png"></div>

<h3><a href="https://github.blog/changelog/2021-04-20-github-actions-control-permissions-for-github_token/" target="_blank" rel="noopener">GitHub Actions: Control permissions for GITHUB_TOKEN | GitHub Changelog</a></h3>

  <p>GitHub Actions: Control permissions for GITHUB_TOKEN</p>


  </article>

  <div class="onebox-metadata">
    
    
  </div>

  <div style="clear: both"></div>
</aside>

<p>But even before that the <code>GITHUB_TOKEN</code> was generally the safer choice, because it’s scoped to the repository. A PAT applies to everything you have access to.</p>
